src/jdk.jpackage/macosx/classes/jdk/jpackage/internal/model/MacPackage.java

```diff
@@ -44,6 +45,36 @@ default Path installDir() {
         return Path.of("/").resolve(relativeInstallDir());
     }
 
+    default boolean isRuntimeImageJDKImage() {
+        if (isRuntimeInstaller()) {
+            Path runtimeImage = app().runtimeImage().orElseThrow();
+            Path p = runtimeImage.resolve("Contents/Home");
+            return !Files.exists(p);
+        }
+
+        return false;
+    }
+
+    // Returns true if signing is requested or JDK bundle is not signed
+    // or JDK image is provided.
+    default boolean isRuntimeJDKBundleNeedSigning() {
+        if (!isRuntimeInstaller()) {
+            return false;
+        }
+
+        if (app().sign()) {
+            return true;
+        }
+
+        if (isRuntimeImageJDKImage()) {
+            return true;
+        } else {
+            Path runtimeImage = app().runtimeImage().orElseThrow();
+            Path p = runtimeImage.resolve("Contents/_CodeSignature");
+            return !Files.exists(p);
+        }
+    }
```

Comment:
Default interface methods should operate on values returned by other interface methods. These new functions involve filesystem operations. They probably don't belong to the interface.